modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterManagementGroupManager.java

```diff
@@ -466,27 +468,51 @@ private CompletableFuture<CmgRaftService> joinCluster(CmgRaftService service, Cl
      * Starts the CMG Raft service using the provided node names as its peers.
      */
     private CompletableFuture<CmgRaftService> startCmgRaftService(Collection<String> nodeNames) {
+        String thisNodeConsistentId = clusterService.topologyService().localMember().name();
+
+        // If we are not in the CMG, we must be a lerner. List of learners will be updated by a leader accordingly,
+        // but just to start a RAFT service we must include ourselves in the initial learners list, that's why we
+        // pass List.of(we) as learners list if we are not in the CMG.
+        List<String> learnerConsistentIds = nodeNames.contains(thisNodeConsistentId) ? List.of() : List.of(thisNodeConsistentId);
+
         try {
             return raftManager
                     .prepareRaftGroup(
                             INSTANCE,
                             nodeNames,
-                            List.of(),
-                            () -> new CmgRaftGroupListener(clusterStateStorage, logicalTopologyService),
+                            learnerConsistentIds,
+                            () -> new CmgRaftGroupListener(clusterStateStorage, logicalTopology, this::onLogicalTopologyChanged),
                             this::createCmgRaftGroupEventsListener,
                             RaftGroupOptions.defaults()
                     )
-                    .thenApply(service -> new CmgRaftService(service, clusterService));
+                    .thenApply(service -> new CmgRaftService(service, clusterService, logicalTopology));
         } catch (Exception e) {
             return failedFuture(e);
         }
     }
 
+    private void onLogicalTopologyChanged(long term) {
+        CompletableFuture<CmgRaftService> serviceFuture = raftService;
```

Comment:
Hm, should we take the `raftServiceLock` here?

Reply:
Good catch. Added synchronization